From cc53a6eb5e4f62303ae7560b3494bfdcc566ddfe Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 3 Dec 2019 11:02:58 -0600
Subject: [PATCH 34/34] ARM64: dts: logicpd-imx8mm-kit: Enable PCIe

There are a few regulators setup to make PCIe work.

The first is a real regulator enabled with GPIO and it's controlled
by the PCIe node directly.

The nCLKREQ pin is not routed to the corresponding pin on the
the SoC, but there is a gpio connected to a GPIO expander. It is used
as a dependent regulator for the real regulator so the regulator sub-
system will auto start this when needed.

Signed-off-by: Adam Ford <adam.ford@logicpd.com>
---
 .../dts/freescale/lpd-imx8mm-baseboard.dtsi   | 21 ++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
index 297058b4653e..ff6ff2d08f2f 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
@@ -39,6 +39,25 @@
 		#reset-cells = <0>;
 	};
 
+	reg_pcie_pwr_en: regulator-pci_pwr_en {
+		compatible = "regulator-fixed";
+		regulator-name = "pci_pwr_en";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		enable-active-high;
+		gpio = <&pca6416_1 1 GPIO_ACTIVE_HIGH>;
+		vin-supply = <&reg_pcie_nclkreq>;
+		startup-delay-us = <5000000>;
+	};
+
+	reg_pcie_nclkreq: regulator-pcie_nclkreq {
+		compatible = "regulator-fixed";
+		regulator-name = "pcie_nclkreq";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&pca6416_1 2 GPIO_ACTIVE_LOW>;
+	};
+
 	reg_usdhc2_vmmc: regulator-usdhc2 {
 		compatible = "regulator-fixed";
 		regulator-name = "VSD_3V3";
@@ -133,7 +152,6 @@
 
 		pinctrl_pcie0: pcie0grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B	0x61 /* open drain, pull up */
 				MX8MM_IOMUXC_GPIO1_IO05_GPIO1_IO5	0x41
 				MX8MM_IOMUXC_SAI2_RXFS_GPIO4_IO21	0x41
 			>;
@@ -356,6 +374,7 @@
 &pcie0{
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie0>;
+	vpcie-supply = <&reg_pcie_pwr_en>;
 	disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio4 21 GPIO_ACTIVE_LOW>;
 	ext_osc = <1>;
-- 
2.17.1

